tests/ha/single_pod_restart/test_server_pod_restart_api.py

```diff
@@ -768,3 +771,159 @@ def test_copy_obj_after_server_pod_restart(self):
+        download_path = os.path.join(self.test_dir_path, "test_chunk_upload" + "_download")
+        chunk_obj_path = os.path.join(self.test_dir_path, self.object_name)
+        upload_op = Queue()
+        workload_info = dict()
+
+        LOGGER.info("Step 1: Perform setup steps for jclient")
+        jc_obj = JCloudClient()
+        resp = self.ha_obj.setup_jclient(jc_obj)
+        assert_utils.assert_true(resp, "Failed in setting up jclient")
+        LOGGER.info("Step 1: Successfully setup jcloud/jclient on runner")
+
+        LOGGER.info("Step 2: Create IAM user with name %s, bucket %s and perform chunk upload",
+                    self.s3acc_name, self.bucket_name)
+        resp = self.rest_obj.create_s3_account(acc_name=self.s3acc_name,
+                                               email_id=self.s3acc_email,
+                                               passwd=S3_CFG["CliConfig"]["s3_account"]["password"])
+        assert_utils.assert_true(resp[0], resp[1])
+        access_key = resp[1]["access_key"]
+        secret_key = resp[1]["secret_key"]
+        self.test_prefix = 'test-44849'
+        self.s3_clean = {'s3_acc': {'accesskey': access_key, 'secretkey': secret_key,
+                                    'user_name': self.s3acc_name}}
+
+        resp = self.ha_obj.create_bucket_chunk_upload(s3_data=self.s3_clean,
+                                                      bucket_name=self.bucket_name,
+                                                      file_size=file_size,
+                                                      chunk_obj_path=chunk_obj_path,
+                                                      background=False)
+        self.extra_files.append(chunk_obj_path)
+        assert_utils.assert_true(resp, "Failure observed in chunk upload in healthy cluster")
+        LOGGER.info("Step 2: Successfully performed chuck upload")
+
+        workload_info[1] = [self.bucket_name, self.object_name]
+
+        LOGGER.info("Calculating checksum of uploaded file %s", chunk_obj_path)
+        upld_chksm_hlt = self.ha_obj.cal_compare_checksum(file_list=[chunk_obj_path],
+                                                          compare=False)[0]
+
+        num_replica = self.num_replica - 1
+        LOGGER.info("Step 3: Shutdown server pod by making replicas=0 and verify cluster & "
+                    "remaining pods status")
+        resp = self.ha_obj.delete_kpod_with_shutdown_methods(
+            master_node_obj=self.node_master_list[0], health_obj=self.hlth_master_list[0],
+            pod_prefix=[const.SERVER_POD_NAME_PREFIX], delete_pod=[self.delete_pod],
+            num_replica=num_replica)
+        # Assert if empty dictionary
+        assert_utils.assert_true(resp[1], "Failed to shutdown/delete pod")
+        pod_name = list(resp[1].keys())[0]
+        if self.set_type == const.STATEFULSET:
+            self.set_name = resp[1][pod_name]['deployment_name']
+        elif self.set_type == const.REPLICASET:
```

Comment:
do we really need this if we are going to have sts only?

Reply:
Ok, we will assume that server and data will be of statefulset.